lib/source-map/source-map-consumer.js

```diff
@@ -305,10 +305,20 @@ define(function (require, exports, module) {
       throw new Error('Unsupported version: ' + version);
     }
 
-    // Some source maps produce relative source paths like "./foo.js" instead of
-    // "foo.js".  Normalize these first so that future comparisons will succeed.
-    // See bugzil.la/1090768.
-    sources = sources.map(util.normalize);
+    sources = sources
+      // Some source maps produce relative source paths like "./foo.js" instead of
+      // "foo.js".  Normalize these first so that future comparisons will succeed.
+      // See bugzil.la/1090768.
+      .map(util.normalize)
+      // Always ensure that absolute sources are internally stored relative to
+      // the source root, if the source root is absolute. Not doing this would
+      // be particularly problematic when the source root is a prefix of the
+      // source (valid, but why??). See github issue #199 and bugzil.la/1188982.
+      .map(function (source) {
+        return sourceRoot && sourceRoot.charAt(0) === '/' && source.charAt(0) === '/'
```

Comment:
you surely meant `charAt(1)`, the second time? those are two times the same check.

but even better would be: `sourceRoot && sourceRoot.startsWith('//')`